### PR TITLE
feat: add simulated delay interceptor

### LINKED ISSUE #12: Feature: Ability to set a simulated delay on responses
What does everyone think about having an option that is passed in the to-be-created config for the `InMemoryDBModule` to set a delay in ms on responses to simulate a real backend server that talks to a DB?

A large use case for this package will be for prototyping, and this seems to be a common need.

### PR DESCRIPTION
**Pull request by jordanpowell88**
Basic implementation of adding a simulated delay via an interceptor.

I noticed while writing e2e test for this scenario that the `npm run e2e` script is NOT actually running the `e2e-spec` files and just `.spec` files.

Waiting on #102 to verify everything is working as expected

Closes #12

### CODE REVIEW

**Comment by wesleygrimes** [comment]
Love the idea of an interceptor! How might we make this work at the feature level? For instance, you have two features: customers and orders. You only want to add a delay to the orders. Is there a way to tie that interceptor to only requests under that feature?

**Comment by jordanpowell88** [comment]
the only way we could use `forFeature` would be to somehow add that as a decorator to the extendable Controller we are building or have the user add the decorator themselves which isn’t ideal. I thought using an interceptor would be cleaner than modifying the `inMemoryDBService` itself. Any other ideas?

File: lib/factories/in-memory-delay-interceptor.factory.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { InMemoryDBConfig } from '../interfaces';
+import { SimulatedDelayInterceptor } from '../interceptors/simulated-delay.interceptor';
+
+export function inMemoryDBDelayInterceptorFactory(
+    featureConfig: Partial<InMemoryDBConfig>
+) {
+    return () => new SimulatedDelayInterceptor(featureConfig);
```
Comment (timdeschryver) [inline lib/factories/in-memory-delay-interceptor.factory.ts]: Nit, should `SimulatedDelayInterceptor` be aware of `InMemoryDBConfig`?
We could introduce a  `NoopInterceptor` which does nothing which saves us a few function calls:

- if `delay` is provided, use `SimulatedDelayInterceptor(config.delay)`
- if `delay` is not provided, use `NoopInterceptor`

Comment (jordanpowell88) [reply lib/factories/in-memory-delay-interceptor.factory.ts]: I like this idea @timdeschryver .

**Comment by jordanpowell88** [comment]
@nestjs-addons/collaborators I'm assuming were going to want to have separate configurable delays by feature. If that is the case I don't think we can go with an `interceptor` even though it is clean and doesn't effect `in-memory-db.service` itself. Anyone else have any ideas?

**Comment by wesleygrimes** [comment]
I think we can still go this route, and then make it come as part of the new Base Controllers. And basically you can pass config through to the controllers and the controllers add the interceptors based on the config.

Basically, if you want the delay feature, then you manually use the interceptor we provide, or you use a base controller.

**Comment by wesleygrimes** [comment]
Also looks like you need to run prettier on this one as the `verify format` step failed.

**Comment by wesleygrimes** [comment]
Moving over to the mono-repo. Can reopen there. https://github.com/nestjs-addons/platform

**Pull request closed without merging**